src/engine/GarbageCollector.ts

```diff
@@ -0,0 +1,82 @@
+export interface GarbageCollectionOptions {
+  textureCollectInterval: number; // default 60_000 ms
+  fontCollectInterval: number; // default 60_000 ms
+  textMeasurementCollectInterval: number; // default 60_000 ms
+}
+
+export interface GarbageCollectorOptions {
+  /**
+   * Returns a timestamp in milliseconds representing now
+   */
+  nowFn: () => number;
+}
+
+
+export class GarbageCollector {
+  private _collectHandle: number;
+  private _running = false;
+  private _collectionMap = new Map<any, [type: string, time: number]>();
+  private _collectors = new Map<string, [(resource: any) => boolean, interval: number]>();
+  constructor(public options: GarbageCollectorOptions) {}
+
+  /**
+   *
+   * @param type Resource type
+   * @param timeoutInterval If resource type exceeds interval in milliseconds collect() is called
+   * @param collect Collection implementation, returns true if collected
+   */
+  registerCollector(type: string, timeoutInterval: number, collect: (resource: any) => boolean) {
+    this._collectors.set(type, [collect, timeoutInterval]);
+  }
+
+  /**
+   * Add a resource to be tracked for collection
+   * @param type
+   * @param resource
+   */
+  addCollectableResource(type: string, resource: any) {
+    this._collectionMap.set(resource, [type, this.options.nowFn()]);
+  }
+
+  /**
+   * Update the resource last used timestamp preventing collection
+   * @param resource
+   */
+  touch(resource: any) {
+    const collectionData = this._collectionMap.get(resource);
+    if (collectionData) {
+      this._collectionMap.set(resource, [collectionData[0], this.options.nowFn()]);
+    }
+  }
+
+  public collect = (deadline?: IdleDeadline) => {
```

Comment:
I can't help but ask a lot of questions here (also applicable for `_collect` in `src/engine/Graphics/Context/texture-loader.ts`)
- What's the reasoning behind making this a property instead of a function?
- In case it stays a property: Is there a reason why it's not marked as `readonly`?

```suggestion
  public collectStaleResources = (deadline?: IdleDeadline) => {
```
(the `collect` verb feels a bit off here as well, but I've no clue what's a better alternative)

Reply:
The reason it's a property is the quircky behavior of `requestIdleCallback` it'll lose it's `this`, I suppose a bind could also work but I do a similar technique for the rAF to close over this without needing to thing about it. 

This technique can mess with subclassing... but I think we'll be okay here.

I do like your name better 👍

Reply:
It totally should be readonly